hardware_interface/src/resource_manager.cpp

```diff
@@ -1610,7 +1618,23 @@ HardwareReadWriteStatus ResourceManager::read(
       auto ret_val = return_type::OK;
       try
       {
-        ret_val = component.read(time, period);
+        if (
+          resource_storage_->hardware_info_map_[component.get_name()].read_rate ==
+          resource_storage_->cm_update_rate_)
+        {
+          ret_val = component.read(time, period);
+        }
+        else
+        {
+          const rclcpp::Duration hw_read_period = rclcpp::Duration::from_seconds(
+            1.0 / resource_storage_->hardware_info_map_[component.get_name()].read_rate);
+          if (
+            component.get_last_read_time().seconds() == 0 ||
+            (time - component.get_last_read_time()).seconds() >= hw_read_period.seconds())
+          {
+            ret_val = component.read(time, hw_read_period);
```

Comment:
Shouldn't we pass the real period instead of the configured one? (no integer divider between hw_read_period and cm_update_rate)

Reply:
Sure. Makes total sense. On it

Reply:
Done. Thanks for pointing out!